### DISCUSSION TITLE
Change route without `<a>` or full reload

### DISCUSSION THREAD

**Discussion by dikkadev** [leptos-rs/leptos#1282] [Q&A] [answered]
Maybe there is something easy to specifically do this and I couldn't find it.
To explain, here is my use-case:

The page looks like this:
```art
┌───────┬─────────┐
│ side  │  work   │
│  bar  │  space  │
└───────┴─────────┘
```

There are 2 routes that are important here:
 - `/` shows the sidebar and workspace with empty/default fields
 - `/snippet/{id}` shows sidebar and workspace loads data for the fields

When going with a `<A>` in the sidebar to each route, everything works nice as expected (just the workspace updates).

Now there is a way (button) to either save what's in the workspace, where we start out at `/` and should go to `/snippet/{id}` with an ID that we get back from a server function. Not knowing the ID in the client until the workspace is saved means I can't have an `<A>` as the button. I need a way to change the route without reloading the whole page (setting `window.location` does a reload). (PS: the sidebar is getting updated by just updating a signal).

There is also a way that goes in return, a delete button. Here we go from a `/snippet/{id}` back to `/`. That could probably be an `<A>`, for which I'd call `preventDefault` is the deleting errors.

All the route changes should of course only occur if the server function does not error.

I apologize for the lengthy explanation, but I'm not sure how else to convey what I'm looking for.

**Comment by gbj** [comment, marked as answer]
Here's an example of how to use [`use_navigate`](https://docs.rs/leptos_router/latest/leptos_router/fn.use_navigate.html) to navigate programmatically. I think the `use_navigate` docs and this example should be plenty to get started on what you're trying to do but feel free to ask if you need more help.

https://github.com/leptos-rs/leptos/blob/39edb6eb45e5969e7b4c1a57650a4af059d7e644/examples/router/src/lib.rs#L195-L210

**Reply by dikkadev** [reply]
Works perfectly, thank you!

**Comment by Miuler** [comment]
Please include the use of use_navigate in the official documentation, this function is widely used
